build/linux/build_linux.sh

```diff
@@ -15,7 +15,7 @@ CORES=$(grep -c ^processor /proc/cpuinfo 2>/dev/null || sysctl -n hw.ncpu)
 rm -r -f $build_folder
 mkdir -p $build_folder
 pushd $build_folder
-cmake -Drun_valgrind:BOOL=ON $build_root -Drun_unittests:BOOL=ON -Drun_int_tests:BOOL=ON
+cmake -Drun_valgrind:BOOL=OFF $build_root -Drun_unittests:BOOL=ON -Drun_int_tests:BOOL=ON
```

Comment:
>OFF [](start = 26, length = 3)

I think we should keep valgrind. Just ssh into the machine and sudo apt-get install it IMO. Won't take long and we'll need it anyways

Reply:
Made a task for this: [https://msazure.visualstudio.com/One/_workitems/edit/8444467](https://msazure.visualstudio.com/One/_workitems/edit/8444467) 

Task 8444467: enable valgrind in azure c pal

I don't think it ever worked. when run_valgrind is set to ON, the valgrind tests will be generated IFF the machine has valgrind installed.

I suspect that the hosted machines DO NOT have valgrind installed, therefore, even you enable it - it doesn't do anything. I can confirm my suispicion here: [https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1604-README.md](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1604-README.md). Valgrind is not there.

And the reason why I disabled it is because OBVIOUSLY we do not pass it.

---
In reply to: [497239807](https://github.com/Azure/azure-c-pal/pull/38#discussion_r497239807) [](ancestors = 497239807)

Reply:
I understand. Sure, separate task is fine.
valgrind was being installed - it was installed through the bash step you removed:

- bash: |
      sudo apt-get update
      sudo apt-get install build-essential pkg-config git cmake uuid-dev valgrind
    displayName: 'Bash Script'

But maybe something else is missing for this repo.
For example ZRPC does use valgrind:

[https://msazure.visualstudio.com/One/_build/results?buildId=35000926&view=logs&j=ae9a2f62-3ee1-5984-579f-87b38be696e5&t=1b4a4e70-15a3-566b-e768-285a68c542e2&l=21](https://msazure.visualstudio.com/One/_build/results?buildId=35000926&view=logs&j=ae9a2f62-3ee1-5984-579f-87b38be696e5&t=1b4a4e70-15a3-566b-e768-285a68c542e2&l=21)

My comment was based on thinking that it does work and it seemed odd that we'd remove it. So separate task.

---
In reply to: [497241016](https://github.com/Azure/azure-c-pal/pull/38#discussion_r497241016) [](ancestors = 497241016,497239807)